lib/yarp/parser_gem_compat.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'ast'
+
+module YARP
+  module ParserGem
+    extend AST::Sexp
+
+    def self.parse(code, filepath = nil)
+      parse_result = YARP.parse(code, filepath)
+      if ENV["DEBUG"]
+        pp parse_result.value
+        puts
+      end
+      convert parse_result.value
+    end
+
+    def self.convert(node)
+      case node
+      when ProgramNode
+        convert node.statements
+      when StatementsNode
+        statements = node.body
+        if statements.size == 1
+          convert statements[0]
+        else
+          s(:begin, *statements.map { convert _1 })
+        end
+      when ArrayNode
+        s(:array, *node.elements.map { convert _1 })
+      when HashNode
+        s(:hash, *node.elements.map { convert _1 })
+      when AssocNode
+        s(:pair, convert(node.key), convert(node.value))
+      when AssocSplatNode
+        s(:kwsplat, convert(node.value))
+      when CallNode
+        arguments = node.arguments ? node.arguments.arguments.map { convert _1 } : []
+        receiver = node.receiver ? convert(node.receiver) : nil
+        s(:send, receiver, node.name.to_sym, *arguments)
```

Comment:
Would be nice if `node.name` returns a Symbol for CallNode.
Maybe `name` fields in general should always be Symbols?